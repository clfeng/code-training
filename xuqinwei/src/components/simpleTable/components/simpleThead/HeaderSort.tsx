import { defineComponent, toRefs } from 'vue'

import { ColumnSortOrder } from '@/types/table'

export default defineComponent({
  name: 'SimpleTableHeaderCell',
  emits: ['sort'],
  props: {
    column: {
      type: Object,
      default: {}
    }
  },
  setup(props, { emit }) {
    let { column } = toRefs(props)

    let onChangeSort = (event: MouseEvent, order) => {
      event.stopPropagation()
      console.log('column.value', column.value);

      emit('sort', column, order)
    }
    return () => {
      return (
        <div class="table__sort">
          {
            /* 升序按钮 */
            column.value.sortOrders?.asc && <span class="asc" onClick={(event: MouseEvent) => onChangeSort(event, ColumnSortOrder.asc)}></span>
          }
          {
            /* 降序按钮 */
            column.value.sortOrders?.desc && <span class="desc" onClick={(event: MouseEvent) => onChangeSort(event, ColumnSortOrder.desc)}></span>
          }
        </div>
      )
    }
  },
})