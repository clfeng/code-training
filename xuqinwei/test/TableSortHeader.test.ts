import { shallowMount, mount } from "@vue/test-utils";
import { expect, test, vi } from "vitest";
import HeaderSort from '../src/components/simpleTable/components/simpleThead/HeaderSort'
test("table header", async () => {
  const onChangeSort = vi.fn()
  const wrapper = shallowMount(HeaderSort, {
    props: {
      column: {
        dataIndex: 3,
        isSort: true,
        label: "年龄",
        order: 2,
        prop: "age",
        sortOrders: { sortOrders: ['normal', 'asc', 'desc'], asc: true, desc: true },
        width: 80
      }
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
  await wrapper.find('.table__sort').findAll('span')[1].trigger('click');
  // onChangeSort
});

