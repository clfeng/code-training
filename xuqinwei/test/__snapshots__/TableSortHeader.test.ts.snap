// Vitest Snapshot v1

exports[`table header 1`] = `
"<div class=\\"table__sort\\">
  <!---->
  <!---->
</div>"
`;
